---
title: "Correlation between epigenetic proteins and physiology"
author: "Jill Ashey"
date: "2024-10-02"
output: html_document
---

Correlation between the epigenetic protein counts from the gene count matrix and the PC1 (calculated in 01-Physiology.Rmd). 

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(tidyverse)
library(readxl)
library(reshape2)
```

Read in gene count matrix. 
```{r}
count_matrix <- read.csv("../data/RNAseq/gene_count_matrix_filtered.csv") %>%
  rename("gene_id" = "X")
head(count_matrix)

# Metadata for RNAseq
rna_meta <- read.csv("../data/RNAseq/metadata.RNAseq.csv")
```
The RNA and physiology samples do not have the same name, so the RNAseq metadata will be needed downstream. 

Read in blast results for Pverr against epigenetic proteins of interest 
```{r}
blast <- read.delim("../output/02-Pver-blast/Mach-blastp-Pver_out.tab", sep = "", header = F)
colnames(blast) <- c("query_id", "subject_id", "percent_identity", "alignment_length", 
                     "mismatches", "gap_openings", "q_start", "q_end", "s_start", 
                     "s_end", "e_value", "bit_score")
head(blast)
```

The gene names in the count matrix and the blast dfs are not the same; this is due to naming errors made by the authors of the [Pverr genome paper](https://academic.oup.com/gbe/article/12/10/1911/5898631?login=false#supplementary-data). They made a supplementary file that has both naming iterations, so this will be used to make sure the gene names are the same in each df 

Read in file with gene name iterations 
```{r}
names <- read_excel("../data/RNAseq/FileS2_Pver_gene_annot_May28.xlsx", skip = 4) %>%
  select(Query, Gene)
names$Gene <- gsub("_gene", "", names$Gene)
```

Join names df with the blast df 
```{r}
blast_names <- blast %>%
  full_join(names, by = c("subject_id" = "Query"))
```

Join blast_names df with count matrix 
```{r}
count_blast <- blast_names %>%
  full_join(count_matrix, by = c("Gene" = "gene_id"))
str(count_blast)
```

Manipulate data so that there is a sample column
```{r}
reshaped_data <- count_blast %>%
  # Pivot longer to stack values
  pivot_longer(cols = C17:E9, 
               names_to = "sample", 
               values_to = "value")
```

Join with RNA metadata 
```{r}
reshaped_data <- reshaped_data %>%
  full_join(rna_meta, by = c("sample" = "sample_id"))
```

Read in phys PC data
```{r}
phys_pc <- read.csv("../data/01-Physiology/phys_data_pc.csv")
```

Join phys data with reshaped data df 
```{r}
count_phys <- phys_pc %>%
  full_join(reshaped_data, by = "fragment.ID") %>%
  na.omit()

count_phys$value <- as.numeric(count_phys$value)
```

Run correlation 
```{r}
correlation_results <- count_phys %>%
  group_by(query_id) %>%
  summarize(
    correlation = list(cor.test(PC1, value))) %>%
  mutate(
    estimate = map_dbl(correlation, ~ .$estimate),
    p.value = map_dbl(correlation, ~ .$p.value)
  ) %>%
  select(query_id, estimate, p.value) %>%
  filter(p.value <= 0.05) %>%
  arrange(desc(estimate)) %>%
  mutate(query_id = factor(query_id, levels = query_id[order(estimate)]))
#correlation_results <- correlation_results[c(443:453),]

head(correlation_results)

write.csv(correlation_results, "../output/03-EpigeneticProteins-Phys-Correlation/epi_protein_correlation.csv")
```

Plot correlation heatmap with the x axis being epigenetic proteins. Only plotting the correlations that have pvalue <= 0.05. 
```{r}
ggplot(correlation_results, aes(x = query_id, y = 1, fill = estimate)) +
  geom_tile(color = "white") +  # Create the tiles
  scale_fill_gradient2(low = "blue", mid = "white", high = "red", midpoint = 0) +  # Color gradient
  geom_text(aes(label = round(estimate, 2)), color = "black") +  # Add text labels
  labs(title = "pâ‰¤0.05",
       x = "Epigenetic Protein",
       y = "") +  # y-axis label empty for a single row
  theme_minimal() +  # Use a minimal theme
  coord_flip() +  # Flip the coordinates so that the Query IDs are on the y-axis
  theme(axis.text.y = element_text(size = 8, color = "black"),  # Display and customize y-axis labels (Query IDs)
        axis.ticks.y = element_blank(),  # Hide y-axis ticks
        axis.text.x = element_blank(),
        axis.title = element_text(size = 10),
        panel.grid.major = element_blank(),  # Remove major grid lines
        panel.grid.minor = element_blank(),  # Remove minor grid lines
        legend.title = element_text(size = 10),  # Customize legend title size
        legend.text = element_text(size = 8))  # Customize legend text size
```

Plot correlations with x axis as PC1 and y axis as gene counts. 
```{r}
sig_prot <- correlation_results$query_id

subset_count_phys <- count_phys %>%
  filter(query_id %in% sig_prot)

ggplot(subset_count_phys, aes(x = PC1, y = value)) + 
  geom_point(size = 3) +
  facet_wrap(~query_id, scales = "free")
```




